queue_job/job.py

```diff
@@ -624,16 +645,12 @@ def delay_from_model(*args, **kwargs):
         "retry_pattern must be a dict"
     )
 
-    if not _is_model_method(func):
-        raise TypeError('@job can only be used on methods of Models')
-
-    inner_func = func.__func__
     delay_func = delay_from_model
 
-    inner_func.delayable = True
-    inner_func.delay = delay_func
-    inner_func.retry_pattern = retry_pattern
-    inner_func.default_channel = default_channel
+    func.delayable = True
+    func.delay = delay_func
```

Comment:
could you replace `delay_func` by `delay_from_model` and remove the initialization of `delay_func` above?